### ISSUE TITLE
Enable Test Mode Per Form

### ISSUE THREAD

**Issue by mehul0810** [WordImpress/Give#2254]
## Issue Overview
Currently, we can enable `Test Mode` globally. However, We can have such feature that a site admin can enable it per form.

I have discussed this point on last Weekly Developer Call and decided to create an issue for further discussion.

**Here are few questions:**
1. Do we have any existing snippet for managing Test Mode Per Form?
2. What do you think this could be a Core feature, Snippet or an Addon?
3. Do we got any Support Requests for this feature till date?
4. What do you think that this feature can be useful?

Let me know your thoughts.

**Comment by ravinderk** [comment]
@DevinWalker I think we can create this as a separate addon.

@mathetos @Benunc @marutim what are your views on this because it will be more useful to you.

**Comment by raftaar1191** [comment]
@ravinderk 

I think it should be in Give Core.

**Comment by ravinderk** [comment]
@raftaar1191 This feature will be used only when an admin or support want to test any payment gateway otherwise no use of it and it will add an extra logic to the core that why I would like to have this feature as an addon which can be used if needed.

Let me know you have any question

**Comment by marutim** [comment]
I have never come across such a requirement so far and nor have I felt a need of one. But I can definitely feel this can be useful for debugging live sites which have multiple donations forms in action and have issue with one specific form. This will definitely be helpful in isolating that one form from others and debugging it in test mode. 

But then I am not sure of the number of such cases coming up in future or has come up in the past. We should wait for Matt's comment on this. He is the guy who has all the **numbers**, :)

**Comment by Benunc** [comment]
It would be an immensely helpful thing to have in my toolbox, to be able to say something like 

> Clone that form using the [Duplicate Posts plugin](https://wordpress.org/plugins/duplicate-post/), and then on the cloned form, enable test mode and see if the error persists.

I think it's useful enough to put into core for a few reasons. 

1. In Give Core, it will receive the benefits of making sure it stays up to date with all payment gateway changes. 
2. It enables us to re-enforce the fact that testing (and sandbox mode, etc) is a best practice if it is in Core.
3. Often staging sites and other methods of testing don't provide the same level of clarity for troublshooting that test mode on the live site does (ssl differences, server inconsistencies, etc). 
4. For larger organizations, putting the live site into test mode is a deal-breaker. They can't afford to take down their site for troubleshooting one gateway. They'll opt for not using the gateway if they have to. This option allows them to test concurrently with a separate test form.

Bottom line: **Yes, I think it is a great idea, and deserves more than a snippet, but doesn't make sense as an addon.**

**Comment by mathetos** [comment]
I agree with Ben about all the reasons this should be in Core. Most importantly, if we're thinking this will help Support, then it helps us a lot less if its an Addon, particularly free support.

**Comment by DevinWalker** [comment]
If it's helpful for support then I think it's a good post-2.0 addition.

**Comment by DevinWalker** [comment]
See comments and discussion on PR about why we ultimately decided this isn't going to be merged into core: https://github.com/WordImpress/Give/pull/2715
